File: src/project/networks/util/TCPPacket.java
package project.networks.util;

public class TCPPacket {
	TCPHeader tcpHeader;
	TCPOptions tcpOptions;
	
	public TCPPacket (TCPHeader tcpHeader_, TCPOptions tcpOptions_) {
		tcpHeader = tcpHeader_;
		tcpOptions = tcpOptions_;
	}
}

File: CPP source/util/TCPPacket.cpp
#include "TCPPacket.h"

TCPPacket::TCPPacket(TCPHeader *tcpHeader_, TCPOptions *tcpOptions_) {
	tcpHeader = tcpHeader_;
	tcpOptions = tcpOptions_;
}

TCPHeader* getTCPHeader() {
	return tcpHeader;
}

void setTCPHeader(TCPHeader* tcpHeader) {
	this.tcpHeader = tcpHeader;
}

TCPOptions* getTCPOptions() {
	return tcpOptions;
}

void setTCPOptions(TCPOptions* tcpOptions) {
	this.tcpOptions = tcpOptions;
}